openedx/core/djangoapps/team_api/tests/test_views.py

```diff
@@ -0,0 +1,437 @@
+        self.student_user_not_active = UserFactory.create(password=self.test_password, is_active=False)
+        self.staff_user = UserFactory.create(password=self.test_password, is_staff=True)
+
+        self.test_team_1 = CourseTeamFactory.create(name='solar team', course_id=self.test_course_1.id, topic_id='renewable')
+        self.test_team_2 = CourseTeamFactory.create(name='Wind Team', course_id=self.test_course_2.id)
+        self.test_team_3 = CourseTeamFactory.create(name='Nuclear Team', course_id=self.test_course_1.id)
+        self.test_team_4 = CourseTeamFactory.create(name='coal team', course_id=self.test_course_2.id, is_active=False)
+
+        self.test_team_1.add_user(self.student_user)
+
+    def setup_inactive_user(self):
+        test_user = UserFactory.create(password=self.test_password)
+        self.client.login(username=test_user.username, password=self.test_password)
+        test_user.is_active = False
+        test_user.save()
+
+    def get_teams_list(self, expected_status, user=None, data=None):
+        user = user if user else self.student_user
+        self.client.login(username=user.username, password=self.test_password)
+        response = self.client.get(reverse('teams_list'), data=data)
+        self.assertEqual(expected_status, response.status_code)
+        return response
+
+    def get_teams_list_json(self, user=None, data=None):
+        response = self.get_teams_list(200, user=user, data=data)
+        return json.loads(response.content)
+
+    def test_list_teams_anonymous(self):
+        response = self.client.get(reverse('teams_list'))
+        self.assertEqual(403, response.status_code)
+
+    def test_list_teams_logged_in(self):
+        teams = self.get_teams_list_json()
+        self.assertEqual(3, teams['count'])
+
+    def test_list_teams_logged_in_not_active(self):
+        self.setup_inactive_user()
+        response = self.client.get(reverse('teams_list'))
+        self.assertEqual(200, response.status_code)
+
+    def test_list_teams_filter_invalid_course_id(self):
+        self.get_teams_list(400, data={'course_id': 'foobar'})
+
+    def test_list_teams_filter_course_id(self):
+        teams = self.get_teams_list_json(data={'course_id': str(self.test_course_2.id)})
+        self.assertEqual(1, teams['count'])
+        self.assertEqual(teams['results'][0]['name'], 'Wind Team')
+
+    def test_list_teams_filter_topic_id(self):
+        teams = self.get_teams_list_json(data={
+            'course_id': str(self.test_course_1.id),
+            'topic_id': 'renewable',
+        })
+        self.assertEqual(1, teams['count'])
+        self.assertEqual(teams['results'][0]['name'], 'solar team')
+
+    def test_list_teams_filter_include_inactive(self):
+        teams = self.get_teams_list_json(data={'include_inactive': True})
+        self.assertEqual(4, teams['count'])
+
+    def test_list_teams_filter_text_search(self):
+        self.get_teams_list(400, data={'text_search': 'foobar'})
```

Comment:
That's fine, but please add an explanatory comments. You probably need doc strings for all of these methods anyway (for pylint/pep8).